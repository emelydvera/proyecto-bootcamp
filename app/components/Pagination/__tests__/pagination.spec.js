const React = require("react");
const Pagination = require("../Pagination");
const { render, screen, fireEvent } = require("@testing-library/react");
const { queryByLabelText, getAllByRole, queryByTitle, debug } = screen;
const UrlGenerator = require("../../../utils/urlGenerator");

describe("Pagination component", () => {
  let urlGenerator = new UrlGenerator();

  it("should render component", () => {
    const { asFragment } = render(
      <Pagination urlGenerator={urlGenerator} totalProducts={10} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render PaginationComponent when totalProducts is smaller than 1000", () => {
    render(<Pagination urlGenerator={urlGenerator} totalProducts={200} />);
    const paginationNav = queryByLabelText("Paginación");
    expect(paginationNav).not.toBeNull();
    expect(paginationNav).toBeInTheDocument();
  });

  it("should render pagination pages when totalProducts is higher than 1000", async () => {
    render(<Pagination urlGenerator={urlGenerator} totalProducts={1002} />);
    const paginationNav = queryByLabelText("Paginación");
    expect(paginationNav).not.toBeNull();
  });

  it("should not render PaginationComponent when totalProducts is smaller than limit", () => {
    render(<Pagination urlGenerator={urlGenerator} totalProducts={2} />);
    const paginationNav = queryByLabelText("Paginación");
    expect(paginationNav).toBeNull();
  });
});
