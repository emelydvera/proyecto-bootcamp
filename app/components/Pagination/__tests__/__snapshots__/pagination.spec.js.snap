// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component should render component 1`] = `<DocumentFragment />`;
